app/models/job.js

```diff
@@ -140,8 +141,14 @@ export default Model.extend(DurationCalculations, {
       return;
     }
     this.set('subscribed', true);
-    if (Travis.pusher) {
-      return Travis.pusher.subscribe('job-' + (this.get('id')));
+
+    if (Travis.pusher && Travis.pusher.ajaxService) {
+      return Travis.pusher.ajaxService.post(Pusher.channel_auth_endpoint, {
+        socket_id: Travis.pusher.pusherSocketId,
+        channels: ['private-job' + this.get('id')]
+      }).then(() => {
+        return Travis.pusher.subscribe('job-' + (this.get('id')));
+      });
```

Comment:
It would be nice if we had a Pusher service to encapsulate interactions with Pusher, as then this could be `this.get('pusher').subscribeToJob(this.get('id'))` or something like that, and all this Pusher-specific code wouldn’t need to pollute the job model. I guess `app/utils/pusher.js` is service-esque, what do you think of moving this code into there?